File: source_code/ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/event_store/sample2.py
import dv_processing as dv
from datetime import timedelta

# Generate 10 events with time range [10000; 20000]
store = dv.data.generate.uniformEventsWithinTimeRange(10000, timedelta(milliseconds=10), (100, 100), 10)

# Get all events beyond and including index 5
events_after_index = store.slice(5)
print(f"1. {events_after_index}")

# Get 3 events starting with index 2
events_in_range = store.slice(2, 3)
print(f"2. {events_in_range}")

# Use sliceBack to retrieve event from the end; this call will retrieve last 3 events
last_events = store.sliceBack(3)
print(f"3. {last_events}")

File: source_code/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/event_store/sample2.cpp
#include <dv-processing/core/core.hpp>
#include <dv-processing/data/generate.hpp>

int main() {
	using namespace std::chrono_literals;

	// Add 10 event with timestamps in range [10000; 20000]
	const auto store = dv::data::generate::uniformEventsWithinTimeRange(10000, 10ms, cv::Size(100, 100), 10);

	// Get all events beyond and including index 5
	dv::EventStore eventsAfterIndex = store.slice(5);
	std::cout << "1. " << eventsAfterIndex << std::endl;

	// Get 3 events starting with index 2
	dv::EventStore eventsInRange = store.slice(2, 3);
	std::cout << "2. " << eventsInRange << std::endl;

	// Use sliceBack to retrieve event from the end; this call will retrieve last 3 events
	dv::EventStore lastEvents = store.sliceBack(3);
	std::cout << "3. " << lastEvents << std::endl;

	return 0;
}
